### PR TITLE
Add `listAuthFactors` method

### PR DESCRIPTION
**Pull request by jonatascastro12**
## Documentation

Does this require changes to the WorkOS Docs? E.g. the [API Reference](https://workos.com/docs/reference) or code snippets need updates.

```
[ ] Yes
```

If yes, link a related docs PR and add a docs maintainer as a reviewer. Their approval is required.

### CODE REVIEW

**Comment by jonatascastro12** [comment]
Current dependencies on/for this PR:
* main
  * **PR #853** <a href="https://app.graphite.dev/github/pr/workos/workos-node/853" target="_blank">Graphite</a> 
    * **PR #854** <a href="https://app.graphite.dev/github/pr/workos/workos-node/854" target="_blank">Graphite</a> 
      * **PR #855** <a href="https://app.graphite.dev/github/pr/workos/workos-node/855" target="_blank">Graphite</a> 
        * **PR #856** <a href="https://app.graphite.dev/github/pr/workos/workos-node/856" target="_blank">Graphite</a> 
          * **PR #857** <a href="https://app.graphite.dev/github/pr/workos/workos-node/857" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/workos/workos-node/857?utm_source=stack-comment).

File: src/users/fixtures/factor.json
Comment (lukemorales) [inline src/users/fixtures/factor.json:1]: nit: can we name the file `auth_factor.json` or `authentication_factor.json`?

File: src/users/users.spec.ts [hunk 1]
```diff
@@ -3,6 +3,7 @@ import MockAdapter from 'axios-mock-adapter';
 import { WorkOS } from '../workos';
 import userFixture from './fixtures/user.json';
 import listUsersFixture from './fixtures/list-users.json';
+import factorFixture from './fixtures/factor.json';
```
Comment (lukemorales) [inline src/users/users.spec.ts:6]:
```suggestion
import authFactorFixture from './fixtures/factor.json';
```

File: src/users/fixtures/factor.json
Comment (jonatascastro12) [reply src/users/fixtures/factor.json:1]: This fixture is following the name from the type which is `Factor`

**Pull request merged**